examples/tomo/xray_trafo.py

```diff
@@ -0,0 +1,217 @@
+
+    # Domain element
+    discr_vol_data = odl.util.phantom.shepp_logan(discr_reco_space, True)
+
+    # Forward projection
+    discr_proj_data = xray_trafo(discr_vol_data)
+
+    # Back projection
+    discr_reco_data = xray_trafo.adjoint(discr_proj_data)
+
+    # Shows a slice of the phantom, projections, and reconstruction
+    discr_vol_data.show(title='parallel 2d volume')
+    discr_proj_data.show(title='parallel 2d sinogram')
+    discr_reco_data.show(title='parallel 2d backprojection')
+
+
+def parallel_3d():
+    # Discrete reconstruction space
+    discr_reco_space = odl.uniform_discr([-20, -20, -20],
+                                         [20, 20, 20],
+                                         [300, 300, 300], dtype='float32')
+
+    # Geometry
+    angle_intvl = odl.Interval(0, 2 * np.pi)
+    dparams = odl.Rectangle([-30, -30], [30, 30])
+    agrid = odl.uniform_sampling(angle_intvl, 360, as_midp=False)
+    dgrid = odl.uniform_sampling(dparams, [558, 558])
+
+    # Astra cannot handle axis aligned origin_to_det unless it is aligned
+    # with the third coordinate axis. See issue #18 at ASTRA's github.
+    # This is fixed in new versions of astra, with older versions, this could
+    # give a zero result.
+    geom = odl.tomo.Parallel3dGeometry(angle_intvl, dparams,
+                                       agrid=agrid, dgrid=dgrid)
+
+    # X-ray transform
+    xray_trafo = odl.tomo.DiscreteXrayTransform(discr_reco_space, geom,
+                                                backend='astra_cuda')
+
+    # Domain element
+    discr_vol_data = odl.util.phantom.shepp_logan(discr_reco_space, True)
+
+    # Forward projection
+    discr_proj_data = xray_trafo(discr_vol_data)
+
+    # Back projection
+    discr_reco_data = xray_trafo.adjoint(discr_proj_data)
+
+    # Shows a slice of the phantom, projections, and reconstruction
+    discr_vol_data.show(indices=np.s_[:, :, 150],
+                        title='parallel 3d volume')
+    discr_proj_data.show(indices=np.s_[0, :, :],
+                         title='parallel 3d projection 0')
+    discr_reco_data.show(indices=np.s_[:, :, 150],
+                         title='parallel 3d backprojection')
+
+
+def fanbeam():
+    # Discrete reconstruction space
+    discr_reco_space = odl.uniform_discr([-20, -20],
+                                         [20, 20],
+                                         [300, 300], dtype='float32')
+
+    # Geometry
+    angle_intvl = odl.Interval(0, 2 * np.pi)
+    dparams = odl.Interval(-30, 30)
+    agrid = odl.uniform_sampling(angle_intvl, 360)
+    dgrid = odl.uniform_sampling(dparams, 558)
+    geom = odl.tomo.FanFlatGeometry(angle_intvl, dparams,
+                                    src_radius=1000, det_radius=100,
+                                    agrid=agrid, dgrid=dgrid)
+
+    # X-ray transform
+    xray_trafo = odl.tomo.DiscreteXrayTransform(discr_reco_space, geom,
+                                                backend='astra_cuda')
```

Comment:
There should be a separate operator like `DiscreteDivergentBeamTransform` for the cone cases. Using `DiscreteXrayTransform` is not correct from the standpoint of mathematical notions.

Related question: Shall we drop the `Discrete` prefix from operators? We don't have non-discrete ones anyway here, and the doc says that it's discrete.